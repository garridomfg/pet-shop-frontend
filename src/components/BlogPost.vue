<template>
  <div class="d-flex">
    <v-responsive>
      <v-img :height="400" cover :src="getImageUrl(props.blog.metadata.image)"></v-img>
    </v-responsive>
    <div class="pa-16 w-50">
      <p class="blog-title mb-4">{{ props.blog.title }}</p>
      <v-btn color="primary" variant="flat">
        <span class="text-white">
          {{ props.blog.title }}
        </span>
      </v-btn>
    </div>
  </div>
</template>
<script setup lang="ts">
import useProducts from '../composables/useProducts'
import { Data } from '../interfaces/products'

const props = defineProps<{
  blog: Data
}>()

const { getImageUrl } = useProducts()
</script>
