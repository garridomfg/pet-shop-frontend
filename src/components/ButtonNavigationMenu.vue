<template>
  <div class="d-flex align-center">
    <v-overlay class="d-flex align-center justify-center" @click:outside="isLogin = true">
      <template #activator="{ props }">
        <v-btn
          v-bind="button.label === 'LOGIN' ? props : null"
          class="mx-3 text-white"
          size="large"
          variant="outlined"
          v-for="button in props.buttons"
          :key="button.label"
          @click="button.fn"
          :prepend-icon="button.icon"
        >
          <template v-if="button.icon ? button.icon : undefined" v-slot:prepend>
            <v-icon color="success"></v-icon>
          </template>
          {{ button.label }}
          <span v-if="button.count >= 0">({{ button.count }})</span>
        </v-btn>
      </template>
      <login-form v-if="isLogin" :toggle-login-register="toggleLoginRegister" />
      <register-form v-else :toggle-login-register="toggleLoginRegister" />
    </v-overlay>
  </div>
</template>
<script setup lang="ts">
import useLogin from '../composables/useLogin'
import { NavButtons } from '../interfaces/navbar'
import { LoginForm, RegisterForm } from '.'

const emit = defineEmits<{
  (e: 'on:toggle-menu', isLogin: boolean): void
}>()

const props = defineProps<{
  buttons: NavButtons[]
}>()

const { isLogin, toggleLoginRegister } = useLogin(emit)
</script>
