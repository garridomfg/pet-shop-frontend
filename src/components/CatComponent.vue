<template>
  <div>
    Cat
  </div>
</template>
<script setup lang="ts">
</script>
<style>
  
</style>