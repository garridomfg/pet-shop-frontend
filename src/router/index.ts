import { createRouter, createWebHistory, type RouteRecordRaw } from 'vue-router'
import NProgress from 'nprogress'
import productsRouter from './products-router'

NProgress.configure({ showSpinner: false })

const routes: RouteRecordRaw[] = [
  {
    path: '/',
    name: 'home',
    component: () => import(/* webpackChunkName: "home-view" */ '@/views/HomeView.vue')
  },
  {
    path: '/products',
    name: 'products',
    ...productsRouter
  },
  {
    path: '/promotions',
    name: 'promotions',
    component: () => import(/* webpackChunkName: "promotion-view" */ '@/views/PromotionView.vue')
  },
  {
    path: '/blog',
    name: 'blog',
    component: () => import(/* webpackChunkName: "blog-view" */ '@/views/BlogView.vue')
  },
  {
    path: '/cart',
    name: 'cart',
    component: () => import(/* webpackChunkName: "blog-view" */ '@/views/CartView.vue')
  },
  {
    path: '/:pathMatch(.*)+',
    name: '404',
    component: () => import(/* webpackChunkName: "404-view" */ '@/views/PageNotFound.vue')
  }
]

const router = createRouter({
  history: createWebHistory(import.meta.env.BASE_URL),
  routes
})

router.beforeEach(async (to, from) => {
  if (to.name) {
    NProgress.start()
  }

  console.log(from)
})

router.afterEach(() => {
  NProgress.done()
})

export default router
