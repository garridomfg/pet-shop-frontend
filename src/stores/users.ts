import { type User, type UserData } from '../interfaces/login'
import { defineStore } from 'pinia'
import shopApi from '@/api/shopApi'

export type UsersStoreState = {
  user: UserData | null | undefined
  error: string | null | undefined
  loading: boolean | null | undefined
}

export const useUsersStore = defineStore('users', {
  state: () =>
    ({
      user: undefined,
      error: undefined,
      loading: undefined
    }) as UsersStoreState,
  getters: {
    currentUser: (state) => state.user,
    errorMessage: (state) => state.error,
    isLoading: (state) => state.loading
  },
  actions: {
    async getUser(): Promise<void> {
      this.loading = true
      this.error = ''
      try {
        const { data } = await shopApi.get<User | undefined>('/user')
        this.user = data?.data
      } catch (error: any) {
        console.log(error)
        this.error = error.response.data.errors.length
          ? error.response.data.errors
          : error.response.data.error
      } finally {
        this.loading = false
      }
    },
    removeUser() {
      this.user = undefined
    }
  }
})
