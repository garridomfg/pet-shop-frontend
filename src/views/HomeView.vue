<template>
  <v-row>
    <div class="v-col v-col-12 mx-auto d-flex flex-column align-center">
      <search-input class="search-bar" @on:input-search="handleSearch" />
      <search-input-list v-if="products.length" :products="products" />
    </div>
    <promotions-carousel
      v-if="promotions.length"
      :promotions="promotions"
      class="v-col v-col-10 mx-auto mt-12"
    />
    <template v-for="(category, index) in categories" :key="category.uuid">
      <product-slider
        :title="category.title"
        :category-id="category.uuid"
        :products="productsByCategory[index]"
        class="v-col v-col-10 mx-auto mt-16"
      />
      <blog-post v-if="blogs.length" :blog="blogs[index]" :class="getBlogClass(index)" />
    </template>
  </v-row>
</template>

<script setup lang="ts">
import { onMounted } from 'vue'
import useProducts from '../composables/useProducts'
import {
  BlogPost,
  ProductSlider,
  PromotionsCarousel,
  SearchInput,
  SearchInputList
} from '../components'

const {
  products,
  categories,
  promotions,
  blogs,
  handleSearch,
  getBlogClass,
  productsByCategory,
  fetchInitialData
} = useProducts()

onMounted(fetchInitialData)
</script>
