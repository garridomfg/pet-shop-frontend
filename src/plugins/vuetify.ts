import { createVuetify } from 'vuetify'
import * as components from 'vuetify/components'
import * as directives from 'vuetify/directives'

export const vuetify = createVuetify({
  components,
  directives,
  theme: {
    themes: {
      light: {
        dark: false,
        colors: {
          primary: '#4ec690',
          secondary: '#f09e00',
          greyscale: '#9fa2b4',
          grey: '#69817b',
          info: '#1976d2',
          mainInfo: '#2196f3',
          success: '#4caf50',
          'border-color': '#0000003b',
          'disabled-bg': 'rgba(0, 0, 0, 0.12)',
          'disabled-text': 'rgba(0, 0, 0, 0.26)',
          'text-secondary': 'rgba(0, 0, 0, 0.54)',
          'text-primary': 'rgba(0, 0, 0, 0.87)',
        }
      },
    },
  },
})