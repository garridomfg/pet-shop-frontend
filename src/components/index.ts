import BlogPost from './BlogPost.vue'
import ButtonNavigationMenu from './ButtonNavigationMenu.vue'
import FooterItemsList from './FooterItemsList.vue'
import NavBar from './NavBar.vue'
import NavigationMenu from './NavigationMenu.vue'
import ProductSlider from './ProductSlider.vue'
import PromotionsCarousel from './PromotionsCarousel.vue'
import SearchInput from './SearchInput.vue'
import SearchInputList from './SearchInputList.vue'
import ShopFooter from './ShopFooter.vue'

export {
  BlogPost,
  ButtonNavigationMenu,
  FooterItemsList,
  NavBar,
  NavigationMenu,
  ProductSlider,
  PromotionsCarousel,
  SearchInput,
  SearchInputList,
  ShopFooter,
}
