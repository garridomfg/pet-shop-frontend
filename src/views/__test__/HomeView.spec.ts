import { beforeEach, describe, expect, test, vi } from 'vitest'
import { mount } from '@vue/test-utils'
import { createRouter, createWebHistory } from 'vue-router'
import { createPinia, setActivePinia } from 'pinia'

import { createVuetify } from 'vuetify'
import * as components from 'vuetify/components'
import * as directives from 'vuetify/directives'

import { routes } from '../../router'
import { useUsersStore } from '../../stores/users'
import HomeView from '../HomeView.vue'

const router = createRouter({
  history: createWebHistory(),
  routes
})
const vuetify = createVuetify({
  components,
  directives
})

const debounce = vi.fn()

describe('Home View tests', () => {
  let wrapper
  beforeEach(() => {
    setActivePinia(createPinia())

    wrapper = mount(HomeView, {
      global: {
        plugins: [vuetify, router],
        directives: {
          debounce
        }
      }
    })
  })

  test('should match with snapshot', () => {
    const usersStore = useUsersStore()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
