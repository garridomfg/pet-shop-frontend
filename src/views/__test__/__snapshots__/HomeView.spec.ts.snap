// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home View tests > should match with snapshot 1`] = `
"<div class=\\"v-row\\">
  <div class=\\"v-col v-col-12 mx-auto d-flex flex-column align-center\\">
    <div class=\\"search-bar\\">
      <div class=\\"v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-text-field\\">
        <!---->
        <div class=\\"v-input__control\\">
          <div class=\\"v-field v-field--center-affix v-field--prepended v-field--variant-outlined v-theme--light v-locale--is-ltr\\">
            <div class=\\"v-field__overlay\\"></div>
            <div class=\\"v-field__loader\\">
              <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
                <!---->
                <div class=\\"v-progress-linear__background bg-primary\\" style=\\"width: 100%;\\"></div>
                <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                  <div class=\\"v-progress-linear__indeterminate\\">
                    <div class=\\"v-progress-linear__indeterminate long bg-primary\\"></div>
                    <div class=\\"v-progress-linear__indeterminate short bg-primary\\"></div>
                  </div>
                </transition-stub>
                <!---->
              </div>
            </div>
            <div class=\\"v-field__prepend-inner\\">
              <!----><i class=\\"mdi-magnify mdi v-icon notranslate v-theme--light v-icon--size-default text-primary\\" aria-hidden=\\"true\\" base-color=\\"primary\\"></i>
            </div>
            <div class=\\"v-field__field\\" data-no-activator=\\"\\">
              <!----><label class=\\"v-label v-field-label\\" for=\\"searchInput\\">
                <!---->Search products
              </label>
              <!----><input size=\\"1\\" type=\\"text\\" id=\\"searchInput\\" aria-describedby=\\"searchInput-messages\\" class=\\"v-field__input\\">
              <!---->
            </div>
            <!---->
            <!---->
            <div class=\\"v-field__outline text-primary\\">
              <div class=\\"v-field__outline__start\\"></div>
              <div class=\\"v-field__outline__notch\\"><label class=\\"v-label v-field-label v-field-label--floating\\" aria-hidden=\\"true\\" for=\\"searchInput\\">
                  <!---->Search products
                </label></div>
              <div class=\\"v-field__outline__end\\"></div>
              <!---->
            </div>
          </div>
        </div>
        <!---->
        <div class=\\"v-input__details\\">
          <transition-group-stub name=\\"slide-y-transition\\" tag=\\"div\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\" class=\\"v-messages\\" role=\\"alert\\" aria-live=\\"polite\\" id=\\"searchInput-messages\\">
            <!---->
          </transition-group-stub>
          <!---->
        </div>
      </div>
    </div>
    <!--v-if-->
  </div>
  <!--v-if-->
</div>"
`;
